frontend/src/Task.js

```diff
@@ -1,22 +1,135 @@
-import { FaTrashAlt } from "react-icons/fa";
+import { useAtom } from "jotai";
+import { useEffect, useRef, useState } from "react";
+import { createPortal } from "react-dom";
+import { FaCaretDown } from "react-icons/fa";
+import { useUpdateTask } from "./api/api.hooks";
+import { openMenuAtom } from "./atoms";
 
-import { Button } from "./Button";
+const badgeStyle = {
+  backgroundColor: "black",
+  color: "white",
+  display: "inline-block",
+  textAlign: "center",
+  width: "20px",
+  height: "20px",
+  marginRight: "10px",
+  fontSize: "10px",
+  borderRadius: "25%",
+};
 
 export const Task = ({ task, deleteTask }) => {
+  const MenuButton = () => {
+    const [openMenu, setOpenMenu] = useAtom(openMenuAtom);
+    const openMenuButton = useRef(null);
+
+    const menuButtonStyle = {
```

Comment:
All the styles here can be moved outside the component, making it cleaner as there is no logic in them.